aleph/authz.py

```diff
@@ -82,69 +90,63 @@ def can(self, collection, action):
             return False
         return collection in self.collections(action)
 
-    def can_stream(self):
+    def can_stream(self) -> bool:
         return self.logged_in
 
-    def can_bulk_import(self):
+    def can_bulk_import(self) -> bool:
         if not self.session_write:
             return False
         return self.logged_in
 
-    def can_write_role(self, role_id):
-        if not self.session_write or role_id is None:
+    def can_write_role(self, role_id: int) -> bool:
+        if not self.session_write:
             return False
         if self.is_admin:
             return True
-        try:
-            return int(role_id) in self.private_roles
-        except (ValueError, TypeError):
-            return False
+        return role_id in self.private_roles
 
-    def can_read_role(self, role_id):
+    def can_read_role(self, role_id: int) -> bool:
         if self.is_admin:
             return True
-        return int(role_id) in self.roles
-
-    def match(self, roles):
-        """See if there's overlap in roles."""
-        roles = ensure_list(roles)
-        if not len(roles):
-            return False
-        return self.roles.intersection(roles) > 0
+        return role_id in self.roles
 
     @property
-    def role(self):
+    def role(self) -> Role:
         return Role.by_id(self.id)
 
     @property
-    def private_roles(self):
+    def private_roles(self) -> Set[int]:
         if not self.logged_in:
             return set()
         return self.roles.difference(Role.public_roles())
 
-    def to_token(self, scope=None, role=None):
-        exp = datetime.utcnow() + timedelta(days=1)
-        payload = {
-            "u": self.id,
-            "exp": exp,
-            "r": list(self.roles),
-            "a": self.is_admin,
-            "b": self.is_blocked,
-        }
-        if scope is not None:
-            payload["s"] = scope
-        if role is not None:
```

Comment:
Having the full role in the token didn't seem to be used anywhere so I removed that.